data/field_meta.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+// FieldMeta attaches structural metadata to each frame
+type FieldMeta struct {
+	// StepSize is the interval between each value.
+	// If no values exist in the values array, this indictes mising data for that interval
+	// Only supports non-zero step sizes
+	StepSize float64 `json:"stepSize,omitempty"`
+
+	// Path is a browsable path on the datasource.
+	StepScale string `json:"stepScale,omitempty"`
+
+	// Aggregation defines the separator pattern to decode a hiearchy. The default separator is '/'.
+	Aggregation string `json:"aggregation,omitempty"`
```

Comment:
sorry, could you describe this in more detail? maybe just an example would help?

Reply:
dooh -- comment was copied :facepalm: will fix.  And not sure "Aggregation" is the best name

Given the timeseries:
```
time: [t1,t2,t3]
value: [v1,v2,v3]
```
what does v2 represent?  Is it the value exactly at t2?  in prometheus it really mean "a value that happened before t2 (and after t1)"

Reply:
what's the use case for this one?

Reply:
it lets us know if we should draw e.g. a 2s-step-wide heatmap tile or histogram bar to the left or right or middle of a "2pm" value in the data. this way the viz accurately shows in which direction the aggregation was done by the backend.

Reply:
In timeseries, this is kinda like the line settings:
![Peek 2022-01-26 13-09](https://user-images.githubusercontent.com/705951/151247410-e9c82c36-e9f2-46f4-b034-2ed970c7a50b.gif)

FYI:  I think we should start with the Step field, and leave the rest off until we know exactly what they should be and that we can actually use them.  But I added a few variations to justify why I think this should exist on an object rather than directly on the field.

The other alternative is to put them in FieldConfig:
Pro:
* no change to existing serialization
* can easily integrate with the UI

Con:
* less clear semantic link